topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,415 @@
+
+    ```kotlin
+    package com.jetbrains.simplelogin.shared
+    
+    expect fun randomUUID(): String
+    ```
+
+2. Create a `Utils.kt` file in `shared/src/androidMain` and provide the `actual` implementation for `randomUUID()` in Android:
+
+    ```kotlin
+    package com.jetbrains.simplelogin.shared
+    
+    import java.util.*
+    actual fun randomUUID() = UUID.randomUUID().toString()
+    ```
+
+3. Create a `Utils.kt` file in `shared/src/iosMain` and provide the `actual` implementation for `randomUUID()` in iOS:
+
+    ```kotlin
+    package com.jetbrains.simplelogin.shared
+    
+    import platform.Foundation.NSUUID
+    actual fun randomUUID(): String = NSUUID().UUIDString()
+    ```     
+
+For Android and iOS, Kotlin will use different platform-specific implementations.
+
+### Run your cross-platform application on Android
+
+Run your cross-platform application for Android to make sure it works.
+
+![Android login application](android-login.png){width=300}
+
+## Make your cross-platform application work on iOS
+
+Once you've made your Android application cross-platform, you can create an iOS application and reuse the shared business logic in it.
+
+1. [Create an iOS project in Xcode](#create-an-ios-project-in-xcode).
+
+2. [Compile the shared module into a framework for the iOS project](#compile-the-shared-module-into-a-framework-for-the-ios-project).
+
+3. [Connect the framework to your iOS project](#connect-the-framework-to-your-ios-project).
+
+4. [Automate iOS project updates](#automate-ios-project-updates).
+
+5. [Connect the shared module to the iOS project](#connect-the-shared-module-to-the-ios-project).
+
+
+### Create an iOS project in Xcode
 
-### Create a KMM shared module
+1. In Xcode, click **File** | **New** | **Project**.
 
-In your Android project, create a KMM shared module for your shared code.
+2. Select a template for an iOS app and click **Next**.
 
-> The KMM Shared Module Wizard is available in Android Studio version 4.1 or higher.
+   ![iOS project template](ios-project-wizard-1.png)
+
+3. As the product name, specify **simpleLoginIOS** and click **Next**.
+
+   ![iOS project settings](ios-project-wizard-2.png)
+
+4. As the location for your project, select the directory that stores your cross-platform application, for example, `kmm-integrate-into-existing-app`.
+
+In Android Studio, you'll get the following structure:
+
+![iOS project in Android Studio](ios-project-in-as.png){width=194}
+
+You can rename the `simpleLoginIOS` directory to `iosApp` for consistency with other top-level directories of your cross-platform project.
+
+![Renamed iOS project directory in Android Studio](ios-directory-renamed-in-as.png){width=194}
+
+### Compile the shared module into a framework for the iOS project
+
+To use Kotlin code in your iOS project, compile shared code into a `.framework`.
+
+* In Android Studio, run the `packForXcode` Gradle task in the Terminal:
```

Comment:
minor: Do we really need a list with one item?

Reply:
MS uses a single-bullet list for one-step procedures. I think it's a good way to highlight.